mhkit/acoustics/analysis.py

```diff
@@ -795,28 +817,27 @@ def _band_sound_pressure_level(
     # Reference value of sound pressure
     reference = 1e-12  # Pa^2, = 1 uPa^2
 
-    band = {}
-    band["center_freq"] = 10 ** np.arange(
-        np.log10(fmin),
-        np.log10(fmax * bandwidth),
-        step=np.log10(bandwidth),
-    )
-    band["lower_limit"] = band["center_freq"] / half_bandwidth
-    band["upper_limit"] = band["center_freq"] * half_bandwidth
-    octave_bins = np.append(band["lower_limit"], band["upper_limit"][-1])
+    octave_bins, band = _create_frequency_bands(octave, fmin, fmax)
 
     # Manual trapezoidal rule to get Pa^2
     pressure_squared = xr.DataArray(
         coords={"time": spsd["time"], "freq_bins": band["center_freq"]},
         dims=["time", "freq_bins"],
     )
     for i, key in enumerate(band["center_freq"]):
-        band_min = octave_bins[i]
-        band_max = octave_bins[i + 1]
-        pressure_squared.loc[{"freq_bins": key}] = np.trapz(
-            spsd.sel(freq=slice(band_min, band_max)),
-            spsd["freq"].sel(freq=slice(band_min, band_max)),
-        )
+        # Min and max band limits
+        band_range = [octave_bins[i], octave_bins[i + 1]]
+
+        # Interpolate between band frequencies if width is narrow
+        x = spsd["freq"].sel(freq=slice(*band_range))
+        if len(x) < 2:
+            spsd_slc = spsd.interp(freq=band_range)
+            x = band_range
```

Comment:
If the dataset is large could this interp lead to performance issues where we would want to warn the user of what was slowing down the code?

Reply:
Good question, I'm not sure.

This if statement should only happen at low frequencies, so in the next commit I'll slice the dataset down before interpolating. Should be faster.